os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c

```diff
@@ -798,72 +798,79 @@ extern int8_t wifi_scan_result_callback(wifi_utils_scan_list_s *scan_list, int s
 
 
 #if SCAN_WITH_SSID
-static void cmd_wifi_scan_with_ssid(int argc, char **argv)
+#define SCAN_BUF_LEN 1000
+int8_t cmd_wifi_scan_with_ssid(trwifi_scan_config_s *scan_config)
 {
 
 	u8 *channel_list = NULL;
 	u8 *pscan_config = NULL;
 
 	char *ssid = NULL;
 	int ssid_len = 0;
-	//Fully scan
-	int scan_buf_len = 500;
-	if (argc == 3 && argv[1] && argv[2]) {
-		ssid = argv[1];
-		ssid_len = strlen((const char *)argv[1]);
-		if ((ssid_len <= 0) || (ssid_len > 32)) {
-			ndbg("\n\rWrong ssid. Length must be less than 32.");
-			goto exit;
+	unsigned char mode = 0;
+	int num_channel = 0;
+
+	int scan_buf_len = SCAN_BUF_LEN;
+	if (!scan_config) {
+		return;
+	}
+	mode = scan_config->scan_mode;
+
+	if (0 == mode) {	//full scan
+		if (!scan_config->ssid) {

```

Comment:
This line means `if scan_config->ssid is NULL`.
So `ssid = strlen(ssid)` will be always 0 and it is always failed. 
I think you need to change it to `if (scan_config->ssid)`.
Please check and update this.